### PR TITLE
Update json_input.py #2030 validation Json Input

### PR DESCRIPTION
**Pull request by davidberenstein1957**
https://github.com/bentoml/BentoML/issues/2030

## Checklist:

- [ ] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by davidberenstein1957** [comment]
`@bentoml.env(infer_pip_packages=True)
@bentoml.artifacts([ModelArtifact("model")])
class ClassifierService(bentoml.BentoService):
    @dataclass
    class MyDataclass:
        text: str
        apply_json: Optional[bool] = None
    @dataclass
    class MyDataclass2:
        number: int
        apply_json: Optional[bool] = None
    @bentoml.api(input=JsonInput(dataclass=MyDataclass), batch=True) 
    def predict(self, parsed_json_list: List[JsonSerializable]) -> List[JsonSerializable]:
        return [self.artifact.model(json['text'], json['apply_json']) for json in parsed_json_list]
    @bentoml.api(input=JsonInput(dataclass=MyDataclass2), batch=True) 
    def predict_2(self, parsed_json_list: List[JsonSerializable]) -> List[JsonSerializable]:
        return [self.artifact.model(json['number'], json['apply_json']) for json in parsed_json_list]
`

File: bentoml/adapters/json_input.py [hunk 1]
```diff
@@ -70,9 +77,15 @@ def extract_user_func_args(
         for task in tasks:
             try:
                 parsed_json = json.loads(task.data)
+                if self._dataclass:
+                    self._dataclass(**parsed_json)
```
Comment (parano) [inline bentoml/adapters/json_input.py:81]: does this support nested dataclass attributes?

**Comment by parano** [comment]
Thanks for the contribution @davidberenstein1957 - the master branch & 0.13.1 version is currently in maintenance mode and we do not plan to accept new features being added at the moment. Our team has switched focus to develop the 1.0 release, in which we've decided to opt for using pydantic for validating JSON input, which seems like a stronger fit for this purpose than dataclass.  You can find the current implementation here: https://github.com/bentoml/BentoML/blob/bentoml-1.0/bentoml/_internal/io_descriptors/json.py

Let me know if you'd be interested in contributing to the development of 1.0!

**Comment by davidberenstein1957** [comment]
For sure! Let me know how I can help. I have done some PRs but it would be cool to Officially contribute to an open source project. 

Regards,
David

**Comment by parano** [comment]
@davidberenstein1957 definitely, feel free to ping me in slack, would love to help you get started with contributing to the project

**Pull request closed without merging**